src/js/jsx/PanelSet.jsx

```diff
@@ -0,0 +1,265 @@
+                return;
+            }
+            
+            var nextState = {};
+            nextState[panelName] = !this.state[panelName];
+
+            this.getFlux().actions.preferences.setPreferences(nextState);
+            this.setState(nextState);
+        },
+
+        componentDidUpdate: function (prevProps, prevState) {
+            // FIXME: Remove this once we ship with libraries always enabled            
+            if (!prevState.librariesEnabled && this.state.librariesEnabled) {
+                this.getFlux().actions.libraries.beforeStartup()
+                    .bind(this)
+                    .then(function () {
+                        this.getFlux().actions.libraries.afterStartup();
+                    });
+            }
+                        
+            if (prevState[UI.LAYERS_LIBRARY_COL] !== this.state[UI.LAYERS_LIBRARY_COL] ||
+                prevState[UI.PROPERTIES_COL] !== this.state[UI.PROPERTIES_COL]) {
+                var payload = {
+                    panelWidth: React.findDOMNode(this.refs.panelSet).clientWidth
+                };
+
+                this.getFlux().actions.ui.updatePanelSizes(payload);
+                this.getFlux().actions.tools.resetSuperselect();
+            }
+        },
+
+        render: function () {
+            var documentIDs = this.state.documentIDs;
+
+            if (this.state.activeDocumentInitialized && documentIDs.size > 0) {
+                var activeDocument = this.state.activeDocument,
+                    documentProperties = this.state.mountedDocumentIDs.map(function (documentID) {
+                        var current = documentID === activeDocument.id,
+                            document = this.getFlux().store("document").getDocument(documentID),
+                            disabled = document && document.unsupported,
+                            panelSetclassNames = classnames({
+                                "panel-set": true,
+                                "panel-set__active": current
+                            }),
+                            panelTabBarClassNames = classnames({
+                                "panel__tab-bar": true,
+                                "panel__tab-bar_visible": !(this.state[UI.LAYERS_LIBRARY_COL] &&
+                                    this.state[UI.PROPERTIES_COL])
+                            });
+
+                        var libraryPanel = this.state.librariesEnabled ? (
+                            <LibrariesPanel
+                                disabled={disabled}
+                                visible={this.state[UI.LIBRARY_PANEL]}
+                                onVisibilityToggle=
+                                    {this._handlePanelVisibilityToggle.bind(this, UI.LIBRARY_PANEL)} />
+                        ) : null;
+                        
+                        return (<div className={panelSetclassNames} key={documentID}>
```

Comment:
Can you bump everything after the opening parenthesis to a new line, thereby saving one step of indentation for the large JSX block below?